aten/src/ATen/native/Distributions.cpp

```diff
@@ -295,4 +295,30 @@ Tensor _s_dirichlet_cpu(const Tensor& alpha, Generator *gen) {
   });
   return ret;
 }
+
+Tensor multinomial_cpu(const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
+  Tensor result = at::empty({0}, self.options().dtype(kLong));
+  multinomial_out_cpu(result, self, n_sample, with_replacement, gen);
+  return result;
+}
+
+Tensor& multinomial_out_cpu(Tensor& result, const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
+  TORCH_CHECK(self.dim() >= 1, "multinomial only supports at least 1D tensor, got: ", self.dim(), "D");
+  TORCH_CHECK(at::isFloatingType(self.scalar_type()), "multinomial only supports floating-point dtypes for input, got: ", self.scalar_type());
```

Comment:
this doesn't look like it supports Half

Reply:
It's based on https://github.com/pytorch/pytorch/blob/d0a525b59248227adbc2a76c9c21ccdb1ce28607/aten/src/TH/generic/THTensorRandom.cpp#L71